packages/payment-detection/src/near-info-retriever.ts

```diff
@@ -1,33 +1,36 @@
 import { PaymentTypes } from '@requestnetwork/types';
-import { Client } from 'pg';
+import { Connection } from 'autobahn';
+
+const NEAR_WEB_SOCKET_URL = 'wss://near-explorer-wamp.onrender.com/ws';
 
 /**
- * Gets a list of transfer events for an address and payment reference
+ * Gets a list of transfer events for a set of Near payment details
  */
 export class NearInfoRetriever {
-  private contractName: ReturnType<typeof getNearContractName>;
-  private connectionString: ReturnType<typeof getIndexerConnectionString>;
+  private nearWebSocketUrl: string;
   /**
+   * @param paymentReference The reference to identify the payment
    * @param toAddress Address to check
    * @param eventName Indicate if it is an address for payment or refund
    * @param network The id of network we want to check
-   * @param paymentReference The reference to identify the payment
    */
   constructor(
+    private paymentReference: string,
     private toAddress: string,
+    private proxyContractName: string,
+    private procedureName: string,
```

Comment:
The computation of these 2 parameters is moved to the `detector` class, to keep the same low-level behavior as existing retrievers.